addons/web/static/src/views/list/list_view.js

```diff
@@ -19,9 +19,13 @@ export const listView = {
         const { ArchParser } = view;
         const { arch, relatedModels, resModel } = genericProps;
         const archInfo = new ArchParser().parse(arch, relatedModels, resModel);
+        const readonly = genericProps.readonly || !archInfo.activeActions?.edit;
+        const editable = (!readonly && archInfo.editable) || false;
 
         return {
             ...genericProps,
+            readonly: readonly,
+            editable: editable,
```

Comment:
I would remove this props which is misleading. The controller receives the archInfo, you can directly read from there (and consider the fact that we might be readonly as well).